main/src/com/google/refine/expr/functions/ToJson.java

```diff
@@ -0,0 +1,113 @@
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,           
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY           
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+package com.google.refine.expr.functions;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Properties;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.refine.expr.EvalError;
+import com.google.refine.grel.ControlFunctionRegistry;
+import com.google.refine.grel.Function;
+import com.google.refine.util.ParsingUtilities;
+
+public class ToJson implements Function {
+
+    @Override
+    /**
+     * Transfor an array of objects into a json array format string
+     * Has an optional format argument, can parse an array of numbers or dates in the given format
+     * @param args the user input
+     *        args[0] is an array of objects
+     *        args[1] is optional, it is a string which indicates the format
+     * @return string representing array of objects in json array format
+     */
+    public Object call(Properties bindings, Object[] args) {
+        String error_message = " accepts an array of objects and one optional format argument";
+        Object result = new EvalError(ControlFunctionRegistry.getFunctionName(this) + error_message);
+        if (args.length >= 1 && args[0] instanceof Object[]) {
+            Object[] array = (Object[]) args[0];
+            int arraySize = array.length;
+            String format = null;
+            // get format
+            if(args.length == 2 && args[1] instanceof String){
+                format = (String) args[1];
+            }
+            // parse format
+            boolean[] formatFlag = new boolean[arraySize];
+            for (int i = 0; i < arraySize; i++){
+                Object element = array[i];
+                if (element instanceof OffsetDateTime && format != null) {
+                    OffsetDateTime odt = (OffsetDateTime) element;
+                    element = (Object) odt.format(DateTimeFormatter.ofPattern(format));
+                    formatFlag[i] = true;
+                }
+                if (element instanceof Number && format != null) {
+                    element = String.format(format, (Number) element);
+                    formatFlag[i] = true;
+                }
+                array[i] = element;
```

Comment:
Here, you are modifying the argument of the function, which is generally not recommended since it can lead to unpredictable behaviours.

Reply:
Thanks for pointing out the hidden problems. I think your previous feature of the formatting option is really good and I want to keep it. But if you think just including the basic functionality of wrapper over the ``ParsingUtilities.mapper.writeValueAsString``  is ok, I can modify it to just keep that.